File: languages/go/client/webhook.go
package client

import (
	"context"
	"errors"

	"github.com/bloock/bloock-sdk-go/v2/internal/bridge"
	"github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"
	"github.com/bloock/bloock-sdk-go/v2/internal/config"
)

// WebhookClient provides functionality for interacting with [Bloock webhooks].
//
// [Bloock webhooks]: https://bloock.com
type WebhookClient struct {
	bridgeClient bridge.BloockBridge
	configData   *proto.ConfigData
}

// NewWebhookClient creates a new WebhookClient with default configuration.
func NewWebhookClient() WebhookClient {
	return WebhookClient{
		bridgeClient: bridge.NewBloockBridge(),
		configData:   config.NewConfigDataDefault(),
	}
}

// NewWebhookClientWithConfig creates a new WebhookClient with the provided configuration.
func NewWebhookClientWithConfig(configData *proto.ConfigData) WebhookClient {
	return WebhookClient{
		bridgeClient: bridge.NewBloockBridge(),
		configData:   configData,
	}
}

// VerifyWebhookSignature verifies the signature of a webhook payload using the provided parameters.
func (c *WebhookClient) VerifyWebhookSignature(payload []byte, header string, secretKey string, enforceTolerance bool) (bool, error) {
	res, err := c.bridgeClient.Webhook().VerifyWebhookSignature(context.Background(), &proto.VerifyWebhookSignatureRequest{
		ConfigData:       c.configData,
		Payload:          payload,
		Header:           header,
		SecretKey:        secretKey,
		EnforceTolerance: enforceTolerance,
	})

	if err != nil {
		return false, err
	}

	if res.Error != nil {
		return false, errors.New(res.Error.Message)
	}

	return res.IsValid, nil
}

File: languages/js/src/client/webhook.ts
import { BloockBridge } from "../bridge/bridge";
import { ConfigData } from "../bridge/proto/config";
import { NewConfigData } from "../config/config";
import { VerifyWebhookSignatureRequest } from "../bridge/proto/webhook";

/**
 * Provides functionality for interacting with [Bloock Webhook service](https://dashboard.bloock.com/login).
 */
export class WebhookClient {
  private bridge: BloockBridge;
  private configData: ConfigData | undefined;

  /**
   * Creates a new WebhookClient with default configuration.
   * @param configData 
   */
  constructor(configData?: ConfigData) {
    this.bridge = new BloockBridge();
    this.configData = NewConfigData(configData);
  }

  /**
   * Verifies the signature of a webhook payload using the provided parameters.
   * @param payload 
   * @param header 
   * @param secretKey 
   * @param enforceTolerance 
   * @returns 
   */
  public async verifyWebhookSignature(
    payload: Uint8Array,
    header: string,
    secretKey: string,
    enforceTolerance: boolean
  ): Promise<boolean> {
    const request = VerifyWebhookSignatureRequest.fromPartial({
      configData: this.configData,
      payload: payload,
      header: header,
      secretKey: secretKey,
      enforceTolerance: enforceTolerance
    });

    return this.bridge
      .getWebhook()
      .VerifyWebhookSignature(request)
      .then(res => {
        if (res.error) {
          throw res.error;
        }
        return res.isValid;
      });
  }
}
